src/ring_theory/polynomial/chebyshev/dickson.lean

```diff
@@ -0,0 +1,187 @@
+/-!
+# Dickson polynomials
+
+The Dickson polynomials are two families of polynomials indexed by `ℕ`,
+with coefficients in a commutative ring `R` depending on an element `a∈R`.
+They are closely related to the Chebyshev polynomials in the case that `a=1`.
+When `a=0` they are just the family of monomials `x ^ n`.
+
+## Main declarations
+
+* `polynomial.dickson₁`: the Dickson polynomials of the first kind.
+* `polynomial.dickson₂`: the Dickson polynomials of the second kind.
+
+## References
+
+* [R. Lidl, G. L. Mullen and G. Turnwald, _Dickson polynomials_][MR1237403]
+
+## TODO
+
+* Move the definition of `lambdashev` from `chebyshev.defs` into this file
+  (or even remove it because it is a special case of the polynomials defined here?).
+* Show that `lambdashev` is equal to the characteristic polynomial of the adjacency matrix of a
+  type A Dynkin diagram.
+-/
+
+noncomputable theory
+
+namespace polynomial
+
+variables {R S : Type*} [comm_ring R] [comm_ring S]
+
+/-- `dickson'` is an auxiliary non-standard function that lets us unify some proofs for `dickson₁`
+and `dickson₂`.
+-/
+noncomputable def dickson' (p₀ : polynomial R) (a : R) : ℕ → polynomial R
+| 0       := p₀
+| 1       := X
+| (n + 2) := X * dickson' (n + 1) - (monomial 0 a) * dickson' n
+
+abbreviation dickson₁ : R → ℕ → polynomial R := dickson' 2
+abbreviation dickson₂ : R → ℕ → polynomial R := dickson' 1
+
+variables (p₀ : polynomial R) (a : R)
+
+@[simp] lemma dickson'_zero : dickson' p₀ a 0 = p₀ := rfl
+@[simp] lemma dickson'_one : dickson' p₀ a 1 = X := rfl
+lemma dickson'_two : dickson' p₀ a 2 = X ^ 2 - monomial 0 a * p₀ :=
+by simp only [dickson', mul_one, pow_two]
+@[simp] lemma dickson'_add_two (n : ℕ) :
+  dickson' p₀ a (n + 2) = X * dickson' p₀ a (n + 1) - (monomial 0 a) * dickson' p₀ a n :=
+by rw dickson'
+
+lemma dickson'_of_two_le (n : ℕ) (h : 2 ≤ n) :
+  dickson' p₀ a n = X * dickson' p₀ a (n - 1) - (monomial 0 a) * dickson' p₀ a (n - 2) :=
+begin
+  obtain ⟨n, rfl⟩ := nat.exists_eq_add_of_le h,
+  rw add_comm,
+  exact dickson'_add_two p₀ a n
+end
+
+variables {R S a}
+
+lemma map_dickson' (f : R →+* S) :
+  ∀ (n : ℕ), map f (dickson' p₀ a n) = dickson' (map f p₀) (f a) n
+| 0       := by simp only [dickson'_zero, map_nat_cast]
+| 1       := by simp only [dickson'_one, map_X]
+| (n + 2) :=
+begin
+  simp only [dickson'_add_two, map_sub, map_mul, map_X, map_monomial],
+  rw [map_dickson' (n + 1), map_dickson' n]
+end
+
+lemma lambdashev_eq_dickson'_two_one_apply :
+∀ (n : ℕ), lambdashev R n = dickson' 2 1 n
+| 0       := by simp only [lambdashev_zero, dickson'_zero]
+| 1       := by simp only [lambdashev_one, dickson'_one]
+| (n + 2) :=
+begin
+  simp only [lambdashev_add_two, dickson'_add_two],
+  rw [lambdashev_eq_dickson'_two_one_apply, lambdashev_eq_dickson'_two_one_apply],
+  change X * dickson₁ (1 : R) (n + 1) - dickson₁ 1 n = X * dickson₁ 1 (n + 1) - 1 * dickson₁ 1 n,
+  rw one_mul
+end
+
+lemma lambdashev_eq_dickson₁_one : lambdashev R = dickson₁ 1 :=
```

Comment:
It's fine with me if you completely remove `lambdashev`. When I wrote this stuff, I was interested in lambda rings. But I don't know the literature that well, and in particular I didn't know about the Dickson polynomials.

But if you would rather do that in a follow-up PR, that's fine with me.

Reply:
I have now completely removed `lambdashev` replacing it with `dickson 1 1`.